ncat/test/ncat-test.pl

```diff
@@ -3084,6 +3084,74 @@ sub socks5_auth {
 		"--max-conns $count --keep-open with exec", ["--keep-open", "--exec", "/bin/cat"], [], $count);
 }
 
+# Tests for zero byte option.
+
+test "-z client with Connect success exit code (tcp)",
+sub {
+        my ($resp, $pid, $code);
+        local $SIG{CHLD} = sub { };
+
+        my ($s_pid, $s_out, $s_in) = ncat($HOST, $PORT, "-l");
+        my ($c_pid, $c_out, $c_in) = ncat($HOST, $PORT, "-z");
+
+        do {
+                $pid = waitpid($s_pid, 0);
+        } while ($pid > 0 && $pid != $s_pid);
+        $pid == $s_pid or die "$pid != $s_pid";
+        $code = $? >> 8;
+        $code == 0 or die "Exit code was $code, not 0";
+};
+kill_children;
+
+test "-z client sends \"\\0\" to server and exits with success exit code (udp)",
+sub {
+        my ($resp, $pid, $code);
+        local $SIG{CHLD} = sub { };
+
+        my ($s_pid, $s_out, $s_in) = ncat($HOST, $PORT, "--udp", "-l");
+        my ($c_pid, $c_out, $c_in) = ncat($HOST, $PORT, "-z", "--udp");
+        $resp = timeout_read($s_out);
+        $resp eq "\0" or die "Server got \"$resp\", not \"\\0\" from client";
+
+        do {
+                $pid = waitpid($c_pid, 0);
+        } while ($pid > 0 && $pid != $c_pid);
+        $pid == $c_pid or die "$pid != $c_pid";
+        $code = $? >> 8;
+        $code == 0 or die "Exit code was $code, not 0";
+};
+kill_children;
+
+test "-z client with connection refused exit code (tcp)",
+sub {
+        my ($pid, $code);
+        local $SIG{CHLD} = sub { };
+
+        my ($c_pid, $c_out, $c_in) = ncat($HOST, $PORT, "-z");
+        do {
+                $pid = waitpid($c_pid, 0);
+        } while ($pid > 0 && $pid != $c_pid);
+        $pid == $c_pid or die;
+        $code = $? >> 8;
+        $code == 1 or die "Exit code was $code, not 1";
+};
+kill_children;
+
+test "-z client with connection refused exit code (udp)",
+sub {
+        my ($pid, $code);
+        local $SIG{CHLD} = sub { };
+
+        my ($c_pid, $c_out, $c_in) = ncat($HOST, $PORT, "-z --udp");
+        do {
+                $pid = waitpid($c_pid, 0);
+        } while ($pid > 0 && $pid != $c_pid);
+        $pid == $c_pid or die;
+        $code = $? >> 8;
+        $code == 2 or die "Exit code was $code, not 2";
```

Comment:
Why is exit code 2 for this? `ncat/util.c` has this comment for the `die` and `bye` functions:

    /* Exit status 2 indicates a program error other than a network error. */

So I think we should try to have the exit code be 1 in this case.